File: blog-admin/src/type/menu.ts
export interface MenuDTO{
    id: number,
    menuName: string,
    path: string,
    icon: string,
    sortOrder: number,
    pathPattern: string,
    component: string,
    menuType: number,
    parentId: number,
}

export interface MenuVO{
    id: number,
    menuName: string,
    path: string,
    icon: string,
    sortOrder: number,
    pathPattern: string,
    component: string,
    menuType: number,
    parentId: number,
    createTime: string,
    isDel: number,
    children: MenuVO[]
}

File: yelingzi-blog/src/main/java/com/yeling/user/domian/entity/Menu.java
package com.yeling.user.domian.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 菜单
 *
 * @author ican
 * @date 2022/11/29 22:10
 **/
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Menu {

    /**
     * 菜单id
     */
    private Integer id;

    /**
     * 父菜单id
     */
    private Integer parentId;

    /**
     * 名称
     */
    private String menuName;

    /**
     * 路由地址
     */
    private String path;

    /**
     * 菜单图标
     */
    private String icon;

    /**
     * 排序
     */
    private Integer sortOrder;
    /**
     * 文件路径
     */
    private String component;
    /**
     * 菜单类型 1目录 2菜单
     */
    private Integer menuType;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

    /**
     * 是否删除
     */
    private Integer isDel;

}